### PR TITLE
PHOENIX-5875 - Optional logging for IndexTool verification

### PR DESCRIPTION
**Pull request by gjacoby126**

### CODE REVIEW

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/IndexToolIT.java [hunk 1]
```diff
@@ -617,6 +618,125 @@ public void testSplitIndex() throws Exception {
         }
     }
 
+    @Test
```
Comment (kadirozde) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/IndexToolIT.java:621]: These tests should move into IndexToolForNonTxGlobalIndexIT after the recent restructuring of IndexToolIT

File: phoenix-core/src/main/java/org/apache/phoenix/mapreduce/index/IndexTool.java [hunk 2]
```diff
@@ -235,6 +272,11 @@ public static IndexVerifyType fromValue(byte[] value) {
     private static final Option END_TIME_OPTION = new Option("et", "endtime",
             true, "End time for indextool rebuild or verify");
 
+    private static final Option DISABLE_LOGGING_OPTION = new Option("dl",
+        "disable-logging", true
+        , "Disable logging of failed verification rows for BEFORE, " +
```
Comment (priyankporwal) [inline phoenix-core/src/main/java/org/apache/phoenix/mapreduce/index/IndexTool.java:277]: So, we cannot disable logging for ONLY mode?

Comment (gjacoby126) [reply phoenix-core/src/main/java/org/apache/phoenix/mapreduce/index/IndexTool.java:277]: @priyankporwal ONLY counts as BEFORE.

File: phoenix-core/src/main/java/org/apache/phoenix/coprocessor/BaseScannerRegionObserver.java [hunk 3]
```diff
@@ -88,6 +88,8 @@
     public static final String INDEX_REBUILD_PAGING = "_IndexRebuildPaging";
     // Index verification type done by the index tool
     public static final String INDEX_REBUILD_VERIFY_TYPE = "_IndexRebuildVerifyType";
+    public static final String INDEX_REBUILD_DISABLE_LOGGING_VERIFY_TYPE =
```
Comment (swaroopak) [inline phoenix-core/src/main/java/org/apache/phoenix/coprocessor/BaseScannerRegionObserver.java:91]: You would need to rebase with latest 4.x I suppose.

File: phoenix-core/src/it/java/org/apache/phoenix/end2end/IndexToolIT.java [hunk 4]
```diff
@@ -617,6 +618,125 @@ public void testSplitIndex() throws Exception {
...
+            assertDisableLogging(conn, 0, IndexTool.IndexVerifyType.BOTH,
+                IndexTool.IndexDisableLoggingType.BOTH,
+                IndexVerificationOutputRepository.PHASE_AFTER_VALUE, schemaName,
+                dataTableName, indexTableName,
+                indexTableFullName, -1);
+
+        }
+    }
+
+    public void truncateTable(Connection conn, TableName tableName) throws SQLException,
```
Comment (gokceni) [inline phoenix-core/src/it/java/org/apache/phoenix/end2end/IndexToolIT.java:697]: Do you want to rename this as deleteAll? It doesn't call truncate as this name suggests

File: phoenix-core/src/main/java/org/apache/phoenix/mapreduce/index/IndexTool.java [hunk 5]
```diff
@@ -320,9 +363,52 @@ public CommandLine parseOptions(String[] args) {
...
+            cmdLine.hasOption(VERIFY_OPTION.getOpt());
+        if (!hasVerifyOption) {
+            return true;
+        }
+        String loggingDisableValue = cmdLine.getOptionValue(DISABLE_LOGGING_OPTION.getOpt());
+        String verifyValue = cmdLine.getOptionValue(VERIFY_OPTION.getOpt());
+        IndexDisableLoggingType loggingDisableType = IndexDisableLoggingType.fromValue(loggingDisableValue);
+        if (loggingDisableType != IndexDisableLoggingType.BEFORE &&
+            loggingDisableType != IndexDisableLoggingType.AFTER &&
+        loggingDisableType != IndexDisableLoggingType.BOTH) {
```
Comment (gokceni) [inline phoenix-core/src/main/java/org/apache/phoenix/mapreduce/index/IndexTool.java:391]: nit:spacing is off

File: phoenix-core/src/main/java/org/apache/phoenix/coprocessor/BaseScannerRegionObserver.java (hunk 3 above)
Comment (gjacoby126) [reply phoenix-core/src/main/java/org/apache/phoenix/coprocessor/BaseScannerRegionObserver.java:92]: Done.

**Comment by gjacoby126** [comment]
A few test flappers and minicluster crashes happened on the Pre-Commit job. Reran them manually locally and they passed, so I'm going to commit this.

**Pull request merged**